exporters/dsexporter/dsexporter.go

```diff
@@ -1,39 +1,106 @@
 package main
 
 import (
+	"context"
+	"encoding/json"
 	"fmt"
 	"net/http"
+	"os"
+	"regexp"
 
 	"github.com/prometheus/client_golang/prometheus"
 	"github.com/prometheus/client_golang/prometheus/promhttp"
+	"k8s.io/client-go/kubernetes"
+	"sigs.k8s.io/controller-runtime/pkg/client/config"
 )
 
+const service = "grafana"
+const check = "prometheus-appstudio-ds"
+
 type CustomCollector struct {
-	requestCounter prometheus.Counter
+	konfluxUp *prometheus.GaugeVec
 }
 
 // Creating a new instance of CustomCollector.
 func NewCustomCollector() *CustomCollector {
 	return &CustomCollector{
-		requestCounter: prometheus.NewCounter(prometheus.CounterOpts{
-			Name: "request_count",
-			Help: "Number of requests handled by the handler",
-		}),
+		konfluxUp: prometheus.NewGaugeVec(prometheus.GaugeOpts{
+			Name: "konflux_up",
+			Help: "Availability of the Konflux default grafana datasource",
+		},
+		[]string{"service", "check"}),
 	}
 }
 
 // Describe method sends descriptions of the metrics to Prometheus.
 // When Prometheus scrapes the /metrics endpoint of the exporter,
 // it first calls the Describe method to get a description of all the metrics.
 func (e *CustomCollector) Describe(ch chan<- *prometheus.Desc) {
-	e.requestCounter.Describe(ch)
+	e.konfluxUp.Describe(ch)
 }
 
 // Collect method sends the current values of the metrics to Prometheus.
 // After Prometheus understands what metrics are available (using the `Describe` method),
 // it then calls the `Collect` method to actually get the values of those metrics.
 func (e *CustomCollector) Collect(ch chan<- prometheus.Metric) {
-	e.requestCounter.Collect(ch)
+	var availability float64
+	dsToCheck := ".*/" + check + "/.*"
```

Comment:
Checking for regex `.*/SOMETHING/.*` is the same as checking if the string contains `/SOMETHING/`. If this is what we're checking then I'd prefer using contains rather than introducing a regex. We should only introduce regex when it provides something that we cannot do without it (as they are hard to maintain).

Reply:
In this case the regex does not add any value and the `contain` is easier to understand

Reply:
Updated to `Contains` again